tests/common/connections/base_console_conn.py

```diff
@@ -81,3 +89,30 @@ def cleanup(self):
     def disconnect(self):
         super(BaseConsoleConn, self).disconnect()
 
+    def posix_shell(self):
```

Comment:
You probably can abstract this posix_shell() method out to a more generic class so that you can use it in your devutil change too?